src/parser/command.rs

```diff
@@ -360,6 +360,14 @@ pub fn parse_set_frequency(input: ParserInput) -> ParserResult<Instruction> {
     Ok((input, Instruction::SetFrequency { frame, frequency }))
 }
 
+/// Parse the contents of a `SET-PHASE` instruction.
+pub fn parse_set_phase(input: ParserInput) -> ParserResult<Instruction> {
+    let (input, frame) = parse_frame_identifier(input)?;
+    let (input, phase) = parse_expression(input)?;
```

Comment:
I'm not totally sure why, but the [Quil-T proposal](https://github.com/quil-lang/quil/blob/master/rfcs/analog/spec_changes.md#phase) requires that the phase be an immediate float (i.e. not a memory reference which `parse_expression` might return).

Reply:
Yeah, the current master branch disagrees with that as well as expressions in set/shift frequency. There is an open but stale PR (https://github.com/quil-lang/quil/pull/46/files) which seeks to address it.

I'm writing under the assumption that the pyQuil implementation is correct, which expects expressions in all three of those locations. I suppose `quilc` should be the real arbiter of truth here...

Reply:
Yes, best to model around pyQuil for now until the Quil spec is fixed.